language/libra-vm/src/errors.rs

```diff
@@ -35,18 +36,22 @@ fn error_split(code: u64) -> (u8, u64) {
 /// Converts particular Move abort codes to specific validation error codes for the prologue
 /// Any non-abort non-execution code is considered an invariant violation, specifically
 /// `UNEXPECTED_ERROR_FROM_KNOWN_MOVE_FUNCTION`
-pub fn convert_normal_prologue_error(error: VMError) -> Result<(), VMStatus> {
+pub fn convert_normal_prologue_error(
+    error: VMError,
+    logger: &impl LogContext,
+) -> Result<(), VMStatus> {
     let status = error.into_vm_status();
     Err(match status {
         VMStatus::Executed => VMStatus::Executed,
         VMStatus::MoveAbort(location, code)
             if location != known_locations::account_module_abort() =>
         {
             let (category, reason) = error_split(code);
+            logger.alert();
             error!(
-                "[libra_vm] Unexpected prologue Move abort: {:?}::{:?} \
-                (Category: {:?} Reason: {:?})",
-                location, code, category, reason
+                logger.clone(),
```

Comment:
Remove all these clones, You shouldn't need to clone this. If you do then I need to fix the macros :)

Reply:
well let's talk about it.....

Reply:
Turns out there's an issue because the trait isn't seen as being implemented for `&impl Trait`. This is something we could fix at a later point though

Reply:
correct, and agree we can and should fix at some point.
However I think the way we have things now should be good enough